HoboVM Preview Features

teochaban opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [ ] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [x] Other, please clarify:
    - Adding to a branch that will introduce new API version

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

jayati-tehri committed:
Update Pricesheet properties for savingsPlan (#24708)

* Update consumption.json

* Update PriceSheet.json

* Update consumption.json

* Update PriceSheet.json

* Update consumption.json

* Update PriceSheet.json

* Update consumption.json

* Update consumption.json

* Update consumption.json

* Update PriceSheet.json

PreetPRG committed:
elasticsan preview refresh microsoft.elasticsan 2022-12-01-preview (#24494)

* Adds base for updating Microsoft.ElasticSan from version preview/2021-11-20-preview to version 2022-12-01-preview

* Updates readme

* Updates API version in new specs and examples

* Added PE related changes

* Added Private Link resource API

* Nit change to correct the path for privateLinkResource API

* Updated Lint errors

* prettier check

* resolved comments & added publicNetworkAccess

* removed publicNetworkAccess & NIT changes

* Updated the list operations

* resolved lintDiff

* proper value for examples and removed value from PECollections

* resolved comments

* spell check resolved

* removing volumeCreateParameter

InbalZim committed:
[Hub Generated] Publish private branch 'InbalZim-storagemover-Microsoft.StorageMover-RPSaaSMaster-2023-07-01-preview' (#24390)

* add or modify files

* Updated `credentials` models according to PR #24318 + reverted unwanted changes to `readme.md`

* Removed URI format from SMB credentials to support it receiving empty string in order to clean-up their value

* Removed extra hierarchy of `AzureStorageFileShareEndpointBaseProperties`

vipray-ms committed:
improve NR Examples (#24636)

* improve NR Examples

* addressing AutoRun Errors

* Fixing Prettier Issue

* Adding NextLink in Max Examples

* Formatting Azure Resource Id

* Updating Format of azure subscriptionIds

ArcturusZhang committed:
update code owner (#24754)

msyyc committed:
Update readme.python.md (#24758)

mikeharder committed:
[TypeSpecValidation] Refactor package.json (#24756)

- Improves upon #24745
- Creates eng/tools/package.json for all EngSys tools
- Moves "postinstall" to eng/tools/TypeSpecValidation/package.json
  - Technically incorrect if the package were ever to be published to a registry
  - In practice, the best way I have found so far to use source-based tools in a TypeScript repo

Commit:
Add resize APIs

Commit:
workflow-bot comments: update breaking changes alternative C to point to alternative B (#24770)

chrjia committed:
Add properties for os patching (#24775)

Co-authored-by: Chris Jiang <chrjia@microsoft.com>

williamzhao87 committed:
[Communication] Job Router Public Preview v2022-07-18 (#23804)

* Updated JR swagger with maxNumberOfJobs

* job scheduling swagger changes

* Enforce DistributionMode.kind to be read-only

* retry with removing Required tag from abstract property for DistributionMode

* WorkerState stopped showing up in generated swagger

* Add reofferTimeUtc to decline offer API

* Prettier

* Prettier

* rename WorkerState to RouterWorkerState

* more fixes in polymorphic types

* Rename decline offer request

* prettier fix

* add example for scheduled job

* cpsell fix

* fix response schema validation error

* address review feedback item round 1

* SDK review changes

* More API changes

* Update distribution mode required properties

* Renamings

* job matching mode added

* prettier fixes

* fix examples refs

* update ref examples

* update swagger spec

* fix enum mismatch

* change operator to labelOperator in classification policy get and update ops example

* Update specification/communication/data-plane/JobRouter/preview/2022-07-18-preview/communicationservicejobrouter.json

Co-authored-by: Liangying.Wei <lianwei@microsoft.com>

* hide redundant communicationerror and communicationerrorresponse during codegen

* revert commit

* fix typo

* test whether removing kind from request works

* revert removing kind from request

* make kind read-only

* run prettier

* try adding properties to empty classes

---------

Co-authored-by: Rajarshi Sarkar 🦄 <rsarkar@microsoft.com>
Co-authored-by: Rajarshi Sarkar <73562869+sarkar-rajarshi@users.noreply.github.com>
Co-authored-by: Liangying.Wei <lianwei@microsoft.com>

mssfang committed:
[OpenAI] Rename properties for Java (#24596)

* renames

* more renames

* testing on the deploymentOrModelName

* revert for no breaking changes

mikeharder committed:
[TypeSpec] Upgrade to 0.46.0 (#24776)

azuremq committed:
Release quota on stop (#24778)

* Release quota on stop

* add ReleaseQuotaOnStop to computeinstance properties

* update

* rm

mssfang committed:
[OpenAI] Update generated autorest (#24788)

- Continuation of #24596

abhargavams committed:
Dynatrace.Observability merge to public repo (#24101)

* Initial Commit

* Updating version in all files

* Removed version from readme

* removing version which are not available from readme

* Resolving PR comments (#24205)

* Features/users/abhargava/merge to public repo (#24230)

* Resolving PR comments

* Features/users/abhargava/merge to public repo (#24233)

* Resolving PR comments

* Responding to the comment in PR review

* Fixed typo

* Features/users/abhargava/merge to public repo (#24253)

* Resolving PR comments

* Responding to the comment in PR review

* Fixed typo

* Added a suppression for Avacado issue

* Adding the required field

* Updating suppression reason

* Updating path for suppression

* Adding suppression for deprecated API

* suppression

* suppress

* Update specification/dynatrace/resource-manager/readme.md

Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

* updating path in suppression

* updating suppression

* updating suppression

* Update specification/dynatrace/resource-manager/readme.md: WIP: fixing suppressions

Iteratively work on fixing suppressions

* Update readme.md: update suppressions

* Update readme.md: update suppressions

* Update readme.md: remove Avocado suppressions

---------

Co-authored-by: kanupriyasinghh <113429657+kanupriyasinghh@users.noreply.github.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>
Co-authored-by: Konrad Jamrozik <kojamroz@microsoft.com>

JainRah committed:
Double Header Fix (#24760)

brandom-msft committed:
[OpenAI.Inference] Adding content filter support (#24652)

* first-pass adding annotation

* FilterResult definitions

* move Filter definitions to common

* cleaning up naming and imports

* adding version headers

* added descriptions to objects

* correct self_harm JSON projection

* Correct to "prompt_annotations"

* "prompt_annotations" for chat, too

* Update specification/cognitiveservices/OpenAI.Inference/models/completions.common.tsp

Co-authored-by: Shawn Fang <45607042+mssfang@users.noreply.github.com>

* Add per-choice content_filter_results to /completions

* tsp compile after merge

* recompile with 0.46.0 after today's update

---------

Co-authored-by: Travis Wilson <35748617+trrwilson@users.noreply.github.com>
Co-authored-by: Shawn Fang <45607042+mssfang@users.noreply.github.com>

lmazuel committed:
Update OpenAI PR assignment (#24794)

* Update OpenAI PR assignment

* Update pull_request_assignment.yml

teochaban commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -862,6 +862,107 @@
+        "VirtualMachineSizes"
+      ],
+      "operationId": "Compute_AllowedResizeSizes",
+      "description": "Returns supported virtual machine sizes for resize",
```
Check if we need to specify that its only for compute instance

teochaban replied:
Change computeinstancestate as well

teochaban replied:
Change return to 202 as LRO

ckairen committed:
[typespec-ci] Automatically detect spec folders to validate (#24629)

weidongxu-microsoft committed:
[fist] mgmt, update service name (#24781)

mikeharder committed:
[typespec-pr] Add pipeline (#24802)

mikeharder committed:
[typespec-ci] Remove path wildcard (#24804)

weshaggard committed:
Update engsys CODEOWNERS (#24808)

trrwilson committed:
Align optionality of content filter categories (#24819)

azuremq commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2875,6 +2879,18 @@
+          "type": "boolean",
+          "description": "Specifies whether this compute instance is pending for reboot to finish os patching."
+        },
+        "patchStatus": {
```
What would be the status when there is no patch?

chrjia replied:
will be null

azuremq commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2875,6 +2879,18 @@
+          "type": "string",
+          "description": "The os patching status."
+        },
+        "latestUpdateTime": {
```
Also please add a flag EnableOsPatching for CI creation

Commit:
Add spec ARM PR review workflow diagram to ARM PR description template. (#24823)

* Add spec PR review workflow diagram to ARM PR description template.

Commit:
Update control_plane_template.md: fix link to diagram (#24828)

jpalvarezl committed:
Added generator config for OpenAI `Functions` for the Java SDK (#24765)

maximrytych-ms committed:
Remove repeatability support for token revocation endpoint (#24589)

dhu5432 committed:
Update clientTYpe (#24503)

KalpeshChavan12 committed:
[Hub Generated] Publish private branch 'dev-maintenance-Microsoft.Maintenance-2023-04-01' (#24048)

* Revert "Revert "[Hub Generated] Publish private branch 'dev-maintenance-Microsoft.Maintenance-2023-04-01' (#23832)" (#23923)"

This reverts commit e6c0e188a74d051a3a44728ce058e85447553392.

* Removed 202 and Async supression.

* Update documentation.

* Updated documentation

---------

Co-authored-by: Kalpesh Chavan <kachavan@microsoft.com>

mikeharder committed:
[package-lock.json] Add names to local dependencies (#24846)

- Command "npm update" removes these names, but "npm install" adds them
- It's probably better to keep the names, since users are more likely to run "npm install"

Commit:
Update ci-fix.md: add API Doc Preview TSG (#24843)

* Update ci-fix.md: add API Doc Preview TSG

kashyap-kunal committed:
Remove unsupported SQL Pool Rename/move operation (#24517)

azuremq committed:
Add enableOSPatching and address comments (#24849)

ishanjindalms committed:
Updated text for property isSoftDeleteFeatureStateEditable (#24713)

athulvijayanmsft committed:
Updating folder structure for Microsoft.Migrate (#24550)

* folders restructure and update readme

* revert removed offazure in readme

thomasricci committed:
Removing readonly flag from top level object (#24848)

sarkar-rajarshi committed:
[ACS JobRouter] Add misssing descriptions (#24842)

* Updated JR swagger with maxNumberOfJobs

* sync with upstream main

* add descriptions

* prettier fix

* add more missing descriptions

* add more description fixes

---------

Co-authored-by: williamzhao87 <williamzhao87@gmail.com>

trrwilson committed:
a couple of representation tweaks for codegen (#24847)

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into teom-hobovm08-01-preview

Commit:
Merge branch 'teom-hobovm08-01-preview' of https://github.com/Azure/azure-rest-api-specs into teom-hobovm08-01-preview

Commit:
Removed pricing not relevant to CI

Commit:
Added Resizing state

Commit:
Change targetVMSize name

Commit:
Revert "Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into teom-hobovm08-01-preview"

This reverts commit 47adff637482e53956bcde7a4331f0638a1ddfbc, reversing
changes made to 4b9165da959a09109ecf96e7e70a940ec3989732.

chrjia committed:
Resolve comments for os patching (#24872)

* Add properties for os patching

* resolve comments

* remove autorestartafterupdates

* update description

* rename property

---------

Co-authored-by: Chris Jiang <chrjia@microsoft.com>

chrjia committed:
Rename osPatchingStatus to upper case (#24954)

* Add properties for os patching

* resolve comments

* remove autorestartafterupdates

* update description

* rename property

* Change property osPatchingStatus to upper case

* rename osPatchingStatus to upper case

---------

Co-authored-by: Chris Jiang <chrjia@microsoft.com>

Commit:
Fix some minor issues

visingla-ms commented:
/azp run

visingla-ms commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -862,6 +862,107 @@
+      ],
+      "operationId": "Compute_AllowedResizeSizes",
+      "description": "Returns supported virtual machine sizes for resize",
+      "x-ms-long-running-operation": true,
```
>"x-ms-long-running-operation": true, [](http://example.com/codeflow?start=6&length=36)

GET operation can't be long running.

teochaban replied:
Fixed

visingla-ms commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -862,6 +862,107 @@
       }
     }
   },
+  "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}/allowedVmSizesForResize": {
```
>allowedVmSizesForResize [](http://example.com/codeflow?start=164&length=23)

The "allowedVmSizesForResize" is not a valid resource type name. One option is to modify your API path as a POST Action e.g. POST /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}/getAllowedVmSizesForResize.

teochaban replied:
Could you explain a bit more about why its not an allowed name?

visingla-ms commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -862,6 +862,107 @@
+          }
+        }
+      ],
+      "responses": {
```
Also include 200 response with the response schema (or return 204 if the response will be empty. Only 202 is not allowed in long running post actions.

teochaban replied:
Fixed

teochaban replied:
This does not make sense for our design

visingla-ms commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2774,6 +2889,33 @@
+          "type": "string",
+          "description": "The os patching status."
+        },
+        "latestPatchTime": {
```
>latestPatchTime [](http://example.com/codeflow?start=9&length=15)

Should this be in datetime format, not string?

chrjia replied:
other property like createdOn, modifiedOn are also defined as string. I'm following the same pattern

visingla-ms commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2774,6 +2889,33 @@
+          "type": "boolean",
+          "description": "Specifies whether this compute instance is pending for reboot to finish os patching."
+        },
+        "scheduledRebootTime": {
```
>scheduledRebootTime [](http://example.com/codeflow?start=9&length=19)

Should this be in datetime format, not string?

chrjia replied:
other property like createdOn, modifiedOn are also defined as string. I'm following the same pattern

visingla-ms commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -862,6 +862,107 @@
       }
     }
   },
+  "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}/allowedVmSizesForResize": {
```
Please define these two new operations under "paths" section, not outside. That's why you are getting Swagger SemanticValidation failed errors.

teochaban replied:
Fixed

Commit:
Fix comments

Commit:
Minor changes

Commit:
Fix more builds

Commit:
200 and nextLink

Commit:
Remove nextlink

Commit:
Try 204 not 200

Commit:
Change GET call to POST

Commit:
Remove 204

Commit:
Add 200 response

Commit:
Change example state

ms-zhenhua commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -860,6 +860,116 @@
+        ],
+        "operationId": "Compute_Resize",
+        "description": "Updates the size of a Compute Instance.",
+        "x-ms-long-running-operation": true,
```
>"x-ms-long-running-operation": true [](http://example.com/codeflow?start=8&length=35)

A LRO Post operation with return schema must have "x-ms-long-running-operation-options" extension enabled.

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/long-running-operations-options-validator.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/long-running-operations-options-validator.md)

teochaban replied:
Well, I don't want to have a return schema. I was forced to add the 200 response because there is a LintDiff rule that says that LRO are required to have a 200 and 202. Ideally I want to not have the 200 return at all. Can I have that and you bypass the rule?

ms-zhenhua replied:
If the LRO-Post does not have a schema, please remove 200 and only define 202. If there is still a lint-diff error, please follow https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process to suppress it.

teochaban replied:
The existing LROs do not have 200 and 202, but there is also not a single suppression defined in the Machine Learning Repo that I could find. I find it hard that this would be the first time that someone needs to add an exception in MachineLearningServices. Am I missing something?

ms-zhenhua replied:
The existing content will not be validated by the lint-diff. Only newly added content will be validated.

teochaban replied:
Removed 200, added suppression

ms-zhenhua commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2005,6 +2116,20 @@
           }
         },
+        "enableOSPatching": {
+          "type": "boolean",
```
>boolean [](http://example.com/codeflow?start=19&length=7)

Consider using string enums
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)

chrjia replied:
It will only have 2 values, other boolean type properties in this file are not enums, I'd prefer to keep it as boolean

ms-zhenhua commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2005,6 +2116,20 @@
+          "x-nullable": true
+        },
+        "releaseQuotaOnStop": {
+          "type": "boolean",
```
>boolean [](http://example.com/codeflow?start=19&length=7)

Consider using string enums
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)

teochaban replied:
It will only have 2 values, other boolean type properties in this file are not enums, I'd prefer to keep it as boolean

ms-zhenhua commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2774,6 +2899,33 @@
+          "description": "Time of the latest os patching."
+        },
+        "rebootPending": {
+          "type": "boolean",
```
>boolean [](http://example.com/codeflow?start=19&length=7)

Consider using string enums
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)

chrjia replied:
It will only have 2 values, other boolean type properties in this file are not enums, I'd prefer to keep it as boolean

ms-zhenhua commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2774,6 +2899,33 @@
+      "type": "object",
+      "description": "Returns metadata about the os patching.",
+      "properties": {
+        "patchStatus": {
```
>patchStatus [](http://example.com/codeflow?start=9&length=11)

can it be enumerated ?

teochaban replied:
Done

chrjia committed:
Add enum for patch status (#25208)

* Add properties for os patching

* resolve comments

* remove autorestartafterupdates

* update description

* rename property

* Change property osPatchingStatus to upper case

* rename osPatchingStatus to upper case

* add enum for patch status

* add enum for patch status

---------

Co-authored-by: Chris Jiang <chrjia@microsoft.com>

Commit:
LRO does not return 200

Commit:
Merge branch 'teom-hobovm08-01-preview' of https://github.com/Azure/azure-rest-api-specs into teom-hobovm08-01-preview

Commit:
Format readme

Commit:
Format readme

Commit:
Format readme

Commit:
Fix readme

msyyc commented on specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json:
```diff
@@ -2774,6 +2893,44 @@
+            "Unknown"
+          ],
+          "x-ms-enum": {
+            "name": "OsPatchingStatus",
```
It is not permitted that name of `x-ms-enum` is same with definition name. Otherwise SDK generation will fail. It is better to rename to another name like: https://github.com/Azure/azure-rest-api-specs/blob/b30cc24a19a4140a0f1362a78c9a9c8a23722edb/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json#L925-L939

chrjia committed:
Rename enum for patchStatus (#25257)

* Add properties for os patching

* resolve comments

* remove autorestartafterupdates

* update description

* rename property

* Change property osPatchingStatus to upper case

* rename osPatchingStatus to upper case

* add enum for patch status

* add enum for patch status

* rename enum's name

* rename enum

---------

Co-authored-by: Chris Jiang <chrjia@microsoft.com>

teochaban commented:
/pr RequestMerge
